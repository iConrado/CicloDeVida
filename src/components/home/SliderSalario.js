import React from 'react';

import { 
  View, 
  Text, } from 'react-native';
import Slider from 'react-native-slider';

import styles from '../functions/styles';
import monetizar from '../functions/monetizar';

export default class SliderSalario extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      tmpSalLiq: props.inicial 
    }; 
  }

  render() {
    return (
      <View style={styles.viewVertical}>
        <Text style={styles.label}>Sal√°rio Bruto:</Text>
        <View style={styles.viewHorizontal}>
            <Slider 
              style={styles.home_slSalLiq}
              minimumValue={0}
              maximumValue={50000}
              step={100}
              minimumTrackTintColor='#14567A'
              thumbTintColor='#14567A'
              value={this.state.tmpSalLiq}
              onValueChange={(value) => this.setState({ tmpSalLiq: value })}
            />
          <Text style={styles.home_txSalLiq}>{monetizar(this.state.tmpSalLiq)}</Text>
        </View>
      </View>
    );
  }
}
