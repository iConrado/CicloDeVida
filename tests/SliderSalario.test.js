import React from 'react';
import { shallow } from 'enzyme';
import SliderSalario from '../src/components/home/SliderSalario';
import Slider from 'react-native-slider';

describe('SliderSalario', () => {
  let ret;
  const retorno = valor => { ret = valor; };
  const app = shallow(<SliderSalario inicial={50} retorno={retorno} />);
  const app2 = shallow(<SliderSalario retorno={retorno} />);

  it('renders correctly', () => {  
    expect(app).toMatchSnapshot();
  });

  it('slider sem o valor inicial', () => {  
    app2.find(Slider).at(0).simulate('valueChange', 30);
    expect(app2.state('tmpSalLiq')).toBe(30);
  });

  it('zera o slider', () => {  
    app.find(Slider).at(0).simulate('valueChange', 0);
    expect(app.state('tmpSalLiq')).toBe(0);
  });

  it('altera o valor do slider', () => {  
    app.find(Slider).at(0).simulate('valueChange', 70);
    expect(app.state('tmpSalLiq')).toBe(70);
  });

  it('retorna valor do slider', () => {  
    app.find(Slider).at(0).simulate('slidingComplete', 70);
    expect(ret).toBe(70);
  });
});
