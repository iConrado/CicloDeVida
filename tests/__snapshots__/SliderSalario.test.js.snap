// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderSalario renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderSalario
    inicial={50}
    retorno={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          style={
            Object {
              "color": "#666",
            }
          }
        >
          Salário Líquido:
        </Text>,
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Slider
            animationType="timing"
            debugTouchArea={false}
            maximumTrackTintColor="#b3b3b3"
            maximumValue={50000}
            minimumTrackTintColor="#14567A"
            minimumValue={0}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={100}
            style={
              Object {
                "height": 40,
                "width": "80%",
              }
            }
            thumbTintColor="#14567A"
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={50}
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": "20%",
              }
            }
          >
            R$ 50
          </Text>
        </View>,
      ],
      "style": Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Salário Líquido:",
          "style": Object {
            "color": "#666",
          },
        },
        "ref": null,
        "rendered": "Salário Líquido:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={50000}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={100}
              style={
                Object {
                  "height": 40,
                  "width": "80%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />,
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "20%",
                }
              }
            >
              R$ 50
            </Text>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationType": "timing",
              "debugTouchArea": false,
              "maximumTrackTintColor": "#b3b3b3",
              "maximumValue": 50000,
              "minimumTrackTintColor": "#14567A",
              "minimumValue": 0,
              "onSlidingComplete": [Function],
              "onValueChange": [Function],
              "step": 100,
              "style": Object {
                "height": 40,
                "width": "80%",
              },
              "thumbTintColor": "#14567A",
              "thumbTouchSize": Object {
                "height": 40,
                "width": 40,
              },
              "value": 50,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "R$ 50",
              "style": Object {
                "alignSelf": "center",
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": "20%",
              },
            },
            "ref": null,
            "rendered": "R$ 50",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Salário Líquido:
          </Text>,
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={50000}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={100}
              style={
                Object {
                  "height": 40,
                  "width": "80%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "20%",
                }
              }
            >
              R$ 50
            </Text>
          </View>,
        ],
        "style": Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Salário Líquido:",
            "style": Object {
              "color": "#666",
            },
          },
          "ref": null,
          "rendered": "Salário Líquido:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={50000}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={100}
                style={
                  Object {
                    "height": 40,
                    "width": "80%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />,
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "20%",
                  }
                }
              >
                R$ 50
              </Text>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animationType": "timing",
                "debugTouchArea": false,
                "maximumTrackTintColor": "#b3b3b3",
                "maximumValue": 50000,
                "minimumTrackTintColor": "#14567A",
                "minimumValue": 0,
                "onSlidingComplete": [Function],
                "onValueChange": [Function],
                "step": 100,
                "style": Object {
                  "height": 40,
                  "width": "80%",
                },
                "thumbTintColor": "#14567A",
                "thumbTouchSize": Object {
                  "height": 40,
                  "width": 40,
                },
                "value": 50,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "R$ 50",
                "style": Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "20%",
                },
              },
              "ref": null,
              "rendered": "R$ 50",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
